tests/bsim/bluetooth/mesh/src/test_advertiser.c

```diff
@@ -580,6 +580,189 @@ static void test_rx_random_order(void)
 	PASS();
 }
 
+static void adv_suspend(void)
+{
+	atomic_set_bit(bt_mesh.flags, BT_MESH_SUSPENDED);
+
+	ASSERT_OK_MSG(bt_mesh_adv_disable(), "Failed to disable advertiser sync");
+}
+
+static void adv_resume(void)
+{
+	atomic_clear_bit(bt_mesh.flags, BT_MESH_SUSPENDED);
+
+	if (!IS_ENABLED(CONFIG_BT_EXT_ADV)) {
+		bt_mesh_adv_init();
+	}
+
+	ASSERT_OK_MSG(bt_mesh_adv_enable(), "Failed to enable advertiser");
+}
+
+static void adv_disable_work_handler(struct k_work *work)
+{
+	adv_suspend();
+}
+
+static K_WORK_DEFINE(adv_disable_work, adv_disable_work_handler);
+
+struct adv_suspend_ctx {
+	bool suspend;
+	int i;
```

Comment:
Could it have a more meaningful name rather than just `i`? I'm pretty sure even you as the test author will have to figure out what `i` means after a couple of months. Something like `adv_inst_num` advertiser instance number or similar.